api/client.go

```diff
@@ -701,7 +701,11 @@ func (p lxdCharmProfiler) LXDProfile() lxdprofile.LXDProfile {
 		return nil
 	}
 	if profiler, ok := p.Charm.(charm.LXDProfiler); ok {
-		return profiler.LXDProfile()
+		profile := profiler.LXDProfile()
+		if profile == nil {
```

Comment:
This dance seems rather fragile - as it is, it is repeated in five places in this PR. It will only take one more use of profiler.LXDProfile to be added, or someone to unintentionally remove this code on the basis that profiler.LXDProfile returns nil and the bug comes back.

Either adding an IsNil (or IsValid) function to the interface, or allowing its functions to be safely callable on nil would be more robust. Another option is to have an isNilProfile function that takes the interface knows about "concrete" types, which can then type switch accordingly.